src/Tests/StepDefinitions/CalculateEarningsForLearningPaymentsStepDefinitions.cs

```diff
@@ -1,60 +1,79 @@
+    }
 
-        firstFundingPeriod.DeliveryPeriods.ShouldHaveCorrectFundingPeriods(numberOfInstalments, firstDeliveryPeriod, firstDeliveryAcademicYear);
-        firstFundingPeriod.DeliveryPeriods.ShouldHaveCorrectFundingCalendarMonths(numberOfInstalments, firstCalendarPeriodMonth, firstCalendarPeriodYear);
+    [Then(@"Earnings generated for each month starting from the first delivery period R(.*)-(.*) and first calendar period (.*)/(.*)")]
+    public void VerifyTheEarningsAreRecordedForEachMonthForTheWholeDuration(short firstDeliveryPeriodMonth, short firstDeliveryPeriodYear, short firstCalendarPeriodMonth, short firstCalendarPeriodYear)
+    {
+        var deliveryPeriods = _context.Get<FundingPeriod>().DeliveryPeriods;
+
+        int numberOfInstallments = deliveryPeriods.Count;
+
+        deliveryPeriods.ShouldHaveCorrectFundingPeriods(numberOfInstallments, firstDeliveryPeriodMonth, firstDeliveryPeriodYear);
+        deliveryPeriods.ShouldHaveCorrectFundingCalendarMonths(numberOfInstallments, firstCalendarPeriodMonth, firstCalendarPeriodYear);
+    }
+
+    //The following 2 methods are for the next set of tickets
+    private DateTime ConvertToDateTimeFormat(String requiredDate)
+    {
+        int month = DateTime.ParseExact(requiredDate.Split('-')[0], "MMM", CultureInfo.CurrentCulture).Month;
+        String yearPart = requiredDate.Split('-')[1];
+        var year = yearPart switch
+        {
+            "CurrentYear" => DateTime.Now.Year,
+            "NextYear" => DateTime.Now.Year + 1,
+            _ => throw new Exception("Unsupported format"),
+        };
+        return new DateTime(01, month, year);
     }
 
-    [Then(@"correct Uln, EmployerId, ProviderId, TransferSenderEmployerId, StartDate, TrainingCode, EmployerType information")]
-    public void ThenCorrectUlnEmployerIdProviderIdTransferSenderEmployerIdStartDateTrainingCodeEmployerTypeInformation()
+    private decimal CalculateOnProgramPaymentBasedOnAgreedPriceAndFundingBand(decimal agreed_price, decimal fundingband_value)
     {
-        _earnings.FundingPeriods.First().Uln.Should().Be(Convert.ToInt64(_apprenticeshipCreatedEvent.Uln));
-        _earnings.FundingPeriods.First().EmployerId.Should().Be(_apprenticeshipCreatedEvent.EmployerAccountId);
-        _earnings.FundingPeriods.First().ProviderId.Should().Be(_apprenticeshipCreatedEvent.UKPRN);
-        _earnings.FundingPeriods.First().TransferSenderEmployerId.Should().Be(_apprenticeshipCreatedEvent.FundingEmployerAccountId);
-        _earnings.FundingPeriods.First().StartDate.Should().Be(_apprenticeshipCreatedEvent.ActualStartDate);
-        _earnings.FundingPeriods.First().TrainingCode.Should().Be(_apprenticeshipCreatedEvent.TrainingCode);
-        _earnings.FundingPeriods.First().EmployerType.Should().Be(_apprenticeshipCreatedEvent.FundingType == FundingType.NonLevy ? EmployerType.NonLevy : EmployerType.Levy);
+        return agreed_price * 0.80m;
```

Comment:
I'd much rather we didn't calculate values in tests, they should be part of the feature.

Reply:
This is just a placeholder one to implement for the following tickets. But this is removed now